storagetest/storagetest.go

```diff
@@ -9,62 +9,80 @@ import (
 
 // Test is Storage conformance test.
 func Test(t *testing.T, storage microstorage.Storage) {
-	key := "test-key"
-	value := "test-value"
+	tests := []struct {
+		key   string
+		value string
+	}{
+		{
+			key:   "test-key",
+			value: "test-value",
+		},
 
-	ok, err := storage.Exists(context.TODO(), key)
-	if err != nil {
-		t.Fatal("expected", nil, "got", err)
-	}
-	if ok {
-		t.Fatal("expected", false, "got", true)
+		// Test a key with a trailing slash.
+		{
+			key:   "test-key/",
+			value: "test-value",
+		},
 	}
 
-	err = storage.Put(context.TODO(), key, value)
-	if err != nil {
-		t.Fatal("expected", nil, "got", err)
-	}
+	for _, test := range tests {
+		key := test.key
+		value := test.value
 
-	// Second Set call should pass.
-	err = storage.Put(context.TODO(), key, value)
-	if err != nil {
-		t.Fatal("expected", nil, "got", err)
-	}
+		ok, err := storage.Exists(context.TODO(), key)
+		if err != nil {
+			t.Fatal("expected", nil, "got", err)
+		}
+		if ok {
+			t.Fatal("expected", false, "got", true)
+		}
 
-	ok, err = storage.Exists(context.TODO(), key)
-	if err != nil {
-		t.Fatal("expected", nil, "got", err)
-	}
-	if !ok {
-		t.Fatal("expected", true, "got", false)
-	}
+		err = storage.Put(context.TODO(), key, value)
+		if err != nil {
+			t.Fatal("expected", nil, "got", err)
+		}
 
-	v, err := storage.Search(context.TODO(), key)
-	if err != nil {
-		t.Fatal("expected", nil, "got", err)
-	}
-	if v != value {
-		t.Fatal("expected", value, "got", v)
-	}
+		// Second Set call should pass.
+		err = storage.Put(context.TODO(), key, value)
+		if err != nil {
+			t.Fatal("expected", nil, "got", err)
+		}
 
-	err = storage.Delete(context.TODO(), key)
-	if err != nil {
-		t.Fatal("expected", nil, "got", err)
-	}
+		ok, err = storage.Exists(context.TODO(), key)
+		if err != nil {
+			t.Fatal("expected", nil, "got", err)
+		}
+		if !ok {
+			t.Fatal("expected", true, "got", false)
+		}
 
-	ok, err = storage.Exists(context.TODO(), key)
-	if err != nil {
-		t.Fatal("expected", nil, "got", err)
-	}
-	if ok {
-		t.Fatal("expected", false, "got", true)
-	}
+		v, err := storage.Search(context.TODO(), key)
+		if err != nil {
+			t.Fatal("expected", nil, "got", err)
+		}
+		if v != value {
+			t.Fatal("expected", value, "got", v)
+		}
 
-	v, err = storage.Search(context.TODO(), key)
-	if !microstorage.IsNotFound(err) {
-		t.Fatal("expected", true, "got", false)
-	}
-	if v != "" {
-		t.Fatal("expected", "empty string", "got", v)
+		err = storage.Delete(context.TODO(), key)
```

Comment:
You shouldn't rely on a fact that `Delete` works correctly. Key names should vary between cases.